### PR TITLE
Add enumeration values for ADC registers on F0x0, F0x1, F0x2 and F0x8

### PR DESCRIPTION
**Pull request by HarkonenBade**

### CODE REVIEW

**Comment by jessebraham** [comment]
LGTM :+1:

File: peripherals/adc/adc_aditf4_v1_1.yaml [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+      OneShot: [0, "DMA one shot mode"]
+      Circular: [1, "DMA circular mode"]
+    DMAEN:
+      Disabled: [0, "DMA mode disabled"]
+      Enabled: [1, "DMA mode enabled"]
+  CFGR2:
+    CKMODE:
+      ADDCLK: [0, "Asynchronous clock mode"]
+      PCLK2: [1, "Synchronous clock mode (PCLK/2)"]
+      PCLK4: [2, "Sychronous clock mode (PCLK/4)"]
```
Comment (adamgreig) [inline peripherals/adc/adc_aditf4_v1_1.yaml]: Should be `PCLK_Div_2` and `PCLK_Div_4` to match what we do elsewhere.

File: peripherals/adc/adc_aditf4_v1_1_RM0091.yaml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# ADC as used on F0x1, F0x2 and F0x8
+
+_include:
+  - "adc_aditf4_v1_1.yaml"
+
+ADC:
+  CFGR1:
+    EXTSEL:
+      TIM2TRGO: [2, "Timer 2 TRGO event"]
```
Comment (adamgreig) [inline peripherals/adc/adc_aditf4_v1_1_RM0091.yaml]: As above, `TIM2_TRGO`.

File: peripherals/adc/adc_aditf4_v1_1.yaml [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+    AWDSGL:
+      AllChannels: [0, "Analog watchdog enabled on all channels"]
+      SingleChannel: [1, "Analog watchdog enabled on a single channel"]
+    DISCEN:
+      Disabled: [0, "Discontinuous mode on regular channels disabled"]
+      Enabled: [1, "Discontinuous mode on regular channels enabled"]
+    AUTOFF:
+      Disabled: [0, "Auto-off mode disabled"]
+      Enabled: [1, "Auto-off mode enabled"]
+    CONT:
```
[the remark this answers is not in the document]
Comment (HarkonenBade) [reply peripherals/adc/adc_aditf4_v1_1.yaml]: WAIT is currently missing from the svd defs.

**Comment by HarkonenBade** [comment]
@adamgreig Yeah, I had done what I could without patching the existing svds as I was unsure how to go about that. I had skipped ADC_CR as I was unsure how to word things for flags that invoke actions rather than just set parameters/cfg options.

**Comment by aurelj** [comment]
@HarkonenBade for patching the svd files, you can add a file in `devices/common_patches/` and use the `_add` or `_modify` instruction to add or modify some fields of the svd. Have a look at `ethernet_mac_regs.yaml` for an example. Then just include your new file from `devices/stm32f0x?.yaml` and you should be done.

**Comment by adamgreig** [comment]
Thanks!

**Pull request merged**